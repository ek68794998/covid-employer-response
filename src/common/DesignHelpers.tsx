import React from "react";

export class DesignHelpers {
	public static materialIcon(iconName: string, className?: string): JSX.Element {
		return <i className={`material-icons ${className || ""}`}>{iconName}</i>;
	}
}
