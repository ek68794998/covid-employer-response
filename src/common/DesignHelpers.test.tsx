/* eslint-disable id-blacklist */

import { DesignHelpers } from "./DesignHelpers";

describe("DesignHelpers", () => {
	test.each([
		[ "add", undefined ],
		[ "add", "" ],
		[ "add", "ClassName" ],
		[ "add", "ClassName OtherClassName" ],
		[ "full_icon", undefined ],
		[ "full_icon", "" ],
		[ "full_icon", "ClassName" ],
		[ "full_icon", "ClassName OtherClassName" ],
	])("creates a material icon for %p with class %p (%#)", (icon: string, className?: string) => {
		expect(DesignHelpers.materialIcon(icon, className)).toMatchSnapshot();
	});
});
