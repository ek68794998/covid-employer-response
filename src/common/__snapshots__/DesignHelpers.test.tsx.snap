// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesignHelpers creates a material icon for "add" with class "" (1) 1`] = `
<i
  className="material-icons"
>
  add
</i>
`;

exports[`DesignHelpers creates a material icon for "add" with class "ClassName OtherClassName" (3) 1`] = `
<i
  className="material-icons ClassName OtherClassName"
>
  add
</i>
`;

exports[`DesignHelpers creates a material icon for "add" with class "ClassName" (2) 1`] = `
<i
  className="material-icons ClassName"
>
  add
</i>
`;

exports[`DesignHelpers creates a material icon for "add" with class undefined (0) 1`] = `
<i
  className="material-icons"
>
  add
</i>
`;

exports[`DesignHelpers creates a material icon for "full_icon" with class "" (5) 1`] = `
<i
  className="material-icons"
>
  full_icon
</i>
`;

exports[`DesignHelpers creates a material icon for "full_icon" with class "ClassName OtherClassName" (7) 1`] = `
<i
  className="material-icons ClassName OtherClassName"
>
  full_icon
</i>
`;

exports[`DesignHelpers creates a material icon for "full_icon" with class "ClassName" (6) 1`] = `
<i
  className="material-icons ClassName"
>
  full_icon
</i>
`;

exports[`DesignHelpers creates a material icon for "full_icon" with class undefined (4) 1`] = `
<i
  className="material-icons"
>
  full_icon
</i>
`;
